### PR TITLE
[Release 0.3] Cherry-pick and update Tomcat analysis

### PR DESCRIPTION
**Pull request by mguetta1**
Tested locally with MTA 7.0.3 - test passed

### CODE REVIEW

File: analysis/tc_tomcat.go [hunk 1]
```diff
@@ -1,134 +1,748 @@
...
+						File:     "/addon/source/example-applications/example-1/src/main/resources/persistence.properties",
+						Line:     2,
+						Message:  "When migrating environments, hard-coded IP addresses may need to be modified or eliminated.",
+						CodeSnip: "jdbc.url=jdbc:oracle:thin:@169.60.225.216:1521/XEPDB1",
+					},
+					{
+						File:     "/addon/source/example-applications/example-1/target/classes/persistence.properties",
+						Line:     2,
+						Message:  "When migrating environments, hard-coded IP addresses may need to be modified or eliminated.",
+						CodeSnip: "jdbc.url=jdbc:oracle:thin:@169.60.225.216:1521/XEPDB1",
```
Comment (mguetta1) [inline analysis/tc_tomcat.go:39]: Hi @eemcmullan @pranavgaikwad 
Could you please confirm this additional incident? (comparing to main branch)
I cannot see this file in the application source code: https://github.com/konveyor/example-applications/tree/main/example-1

Comment (mguetta1) [reply analysis/tc_tomcat.go:39]: Selection_581
Selection_582

File: analysis/tc_tomcat.go [hunk 2]
```diff
@@ -1,134 +1,748 @@
...
 					{
-						File:    "persistence.properties",
-						Line:    2,
-						Message: "Hard-coded IP: 169.60.225.216When migrating environments, hard-coded IP addresses may need to be modified or eliminated.",
+						File:     "/addon/source/example-applications/example-1/src/main/resources/persistence.properties",
+						Line:     2,
+						Message:  "When migrating environments, hard-coded IP addresses may need to be modified or eliminated.",
+						CodeSnip: "jdbc.url=jdbc:oracle:thin:@169.60.225.216:1521/XEPDB1",
+					},
+					{
```
Comment (shawn-hurley) [inline analysis/tc_tomcat.go:35]: This is actually showing the other bug that we still need to fix, that the builtin provider for java needs to ignore target folder.

But for now this is correct.

**Review by shawn-hurley — APPROVED** [review]
Looks correct to me, but we need to re-run CI as main is failing nightly

**Review by eemcmullan — APPROVED** [review]

**Pull request merged**